deltacat/utils/ray_utils/retry_handler/batch_scaling_interface.py

```diff
@@ -0,0 +1,31 @@
+from typing import List, Any, Protocol
+from deltacat.utils.ray_utils.retry_handler.task_info_object import TaskInfoObject
+class BatchScalingInterface(Protocol):
+    """
+    Interface for a generic batch scaling that the client can provide.
+    """
+    def has_next_batch(self) -> bool:
+        """
+        Returns true if there are tasks remaining in the overall List of tasks to create a new batch
+        """
+        pass
+    def next_batch(self, task_info: TaskInfoObject) -> List:
+        """
+        Gets the next batch to execute on
+        """
+        pass
+    def mark_task_complete(self, task_info: TaskInfoObject) -> None:
+        """
+        If the task has been completed, mark some field of it as true
+        so we know what tasks are completed and what need to be executed
+        """
+        pass
+
+    def mark_batch_failed(self, task_info: TaskInfoObject) -> None:
+        """
+        If the task returns the exeption that was caught, we mark the task as failed and
+        decrease multiplicative
```

Comment:
multiplicative decrease is the implementation detail. It can be omitted from the interface doc.